esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -562,7 +565,12 @@ void WsWuInfo::getHelpers(IEspECLWorkunit &info, unsigned long flags)
                 getHelpFiles(query, (WUFileType) i, helpers, flags, helpersCount);
         }
 
-        getWorkunitThorLogInfo(helpers, info, flags, helpersCount);
+        SCMStringBuffer zapName;
+        cw->getDebugValue("zapname", zapName);
+        if (zapName.length())
+            getZAPWorkunitThorLogInfo(zapName.str(), helpers, info, flags, helpersCount);
+        else
+            getWorkunitThorLogInfo(helpers, info, flags, helpersCount);
```

Comment:
I haven't followed the logic fully, but this doesn't feel right.
Looks like there's a lot of code special casing the import case, e.g. readZAPWUNumberOfSlaves(), getZAPWorkunitThorLogInfo()

Once imported I wouldn't expect there to be any difference here.
The associated files should have been imported (copied form the zap) and linked to by the xml, in the same way as any other workunit.

Reply:
I may get the idea. Let me try if I can implement.